isis/src/base/objs/CSMCamera/CSMCamera.cpp

```diff
@@ -383,28 +380,28 @@ namespace Isis {
   // CSMCamera(cube) -> Camera(cube) -> Sensor(cube) -> Spice(cube)
   // Spice::init() creates a Target in such a way that requires spice data / tables, so
   // this works around that.
-  Target *CSMCamera::setTarget(Pvl label) {
+  void CSMCamera::setTarget(Pvl label) {
     Target *target = new Target();
     PvlGroup &inst = label.findGroup("Instrument", Pvl::Traverse);
     QString targetName = inst["TargetName"][0];
     target->setName(targetName);
 
     // get radii from CSM
-    csm::SettableEllipsoid *ellipsoidModel = dynamic_cast<csm::SettableEllipsoid*>(m_model);
-    if (!ellipsoidModel) {
-      // TODO is there a fallback we can do here?
-      QString msg = "Failed to get ellipsoid from CSM Model.";
-      throw IException(IException::Programmer, msg, _FILEINFO_);
-    }
-    csm::Ellipsoid targetEllipsoid = ellipsoidModel->getEllipsoid();
-    std::vector<Distance> radii {Distance(targetEllipsoid.getSemiMajorRadius(), Distance::Meters),
-                                 Distance(targetEllipsoid.getSemiMajorRadius(), Distance::Meters),
-                                 Distance(targetEllipsoid.getSemiMinorRadius(), Distance::Meters)};
+    csm::Ellipsoid targetEllipsoid = csm::SettableEllipsoid::getEllipsoid(m_model);
```

Comment:
Does this throw its own error if it fails and that's why the ISIS exception could be removed?

Reply:
This returns the WGS84 ellipsoid if it fails to convert the model to a settableellipsoid. So, this won't raise an exception.